src/AlertsManager.cpp

```diff
@@ -671,13 +678,22 @@ const char* AlertsManager::getAlertType(AlertType alert_type) {
 }
 
 /* **************************************************** */
+SlackNotificationChoice AlertsManager::getSlackNotificationChoice(char* choice) {
+  
+  if (strcmp(choice, "only_errors")==0) return notify_errors_only;
+  if (strcmp(choice, "errors_and_warnings")==0) return notify_errors_and_warnings;
+  return notify_all_alerts; /* default choice*/
+}
+
+
+/* **************************************************** */
 
 void AlertsManager::notifyAlert(AlertEntity alert_entity, const char *alert_entity_value,
 				AlertType alert_type, AlertLevel alert_severity,
 				const char *alert_json,
 				const char *alert_origin, const char *alert_target) {
   json_object *notification;
-  char alert_sender_name[64], message[2015];
+  char alert_sender_name[64], message[2015], notification_username[96];
```

Comment:
don't hardcode sizes here. Look in ntop_defines and look for already defined constants such as https://github.com/SauroPollastrini/ntopng/blob/dev/include/ntop_defines.h#L277